libraries/SPI/SPI.cpp

```diff
@@ -565,18 +566,58 @@ void SPIClass::transferBytes_(const uint8_t * out, uint8_t * in, uint8_t size) {
     SPI1CMD |= SPIBUSY;
     while(SPI1CMD & SPIBUSY) {}
 
-    if(in) {
-        uint32_t * dataPtr = (uint32_t*) in;
+    if (in) {
+        uint32_t *dataPtr = (uint32_t*) in;
         fifoPtr = &SPI1W0;
-        dataSize = ((size + 3) / 4);
-        while(dataSize--) {
-            *dataPtr = *fifoPtr;
-            dataPtr++;
-            fifoPtr++;
+        dataSize = size;
```

Comment:
dataSize = size here, but it was (size+3)/4 before. From the below code, I suspect it's to handle the last 1-3 bytes at the end when size is not a multiple of 4.
If that is correct, why is the above code for out the same as before? Shouldn't that be changed in the same way?

Reply:
Renamed and commented for clarity.